coderd/searchquery/search.go

```diff
@@ -277,15 +278,16 @@ func Templates(ctx context.Context, db database.Store, actorID uuid.UUID, query
 
 	parser := httpapi.NewQueryParamParser()
 	filter := database.GetTemplatesWithFilterParams{
-		Deleted:        parser.Boolean(values, false, "deleted"),
-		OrganizationID: parseOrganization(ctx, db, parser, values, "organization"),
-		ExactName:      parser.String(values, "", "exact_name"),
-		FuzzyName:      parser.String(values, "", "name"),
-		IDs:            parser.UUIDs(values, []uuid.UUID{}, "ids"),
-		Deprecated:     parser.NullableBoolean(values, sql.NullBool{}, "deprecated"),
-		HasAITask:      parser.NullableBoolean(values, sql.NullBool{}, "has-ai-task"),
-		AuthorID:       parser.UUID(values, uuid.Nil, "author_id"),
-		AuthorUsername: parser.String(values, "", "author"),
+		Deleted:          parser.Boolean(values, false, "deleted"),
+		OrganizationID:   parseOrganization(ctx, db, parser, values, "organization"),
+		ExactName:        parser.String(values, "", "exact_name"),
+		FuzzyName:        parser.String(values, "", "name"),
+		IDs:              parser.UUIDs(values, []uuid.UUID{}, "ids"),
+		Deprecated:       parser.NullableBoolean(values, sql.NullBool{}, "deprecated"),
+		HasAITask:        parser.NullableBoolean(values, sql.NullBool{}, "has-ai-task"),
+		AuthorID:         parser.UUID(values, uuid.Nil, "author_id"),
+		AuthorUsername:   parser.String(values, "", "author"),
+		HasExternalAgent: parser.NullableBoolean(values, sql.NullBool{}, "has-external-agent"),
```

Comment:
This mixes `-` and `_`. I know `has-ai-task` uses dashes, but it probably needs to be changed to underscores...